media/js/firefox/welcome/welcome8.js

```diff
@@ -2,9 +2,72 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-(function() {
+(function (Mozilla) {
     'use strict';
 
+    /* update dataLayer with experiment info */
+    var href = window.location.href;
+
+    var initTrafficCop = function () {
+        if (href.indexOf('v=') !== -1) {
+            var fbContainerLink = document.getElementById('fb-container-link');
+            var monitorLink =  document.getElementById('monitor-link');
+            var utmSource = '';
+            if (href.indexOf('v=text') !== -1) {
+                utmSource = 'mozilla.org-firefox-welcome-8-text';
+                // default, no special class needed
+                window.dataLayer.push({
+                    'data-ex-variant': 'experiment-hvt-visual-text',
+                    'data-ex-name': 'experiment-hvt-visual'
+                });
+            } else if (href.indexOf('v=image') !== -1) {
+                utmSource = 'mozilla.org-firefox-welcome-8-image';
+                document.body.classList.add('variation-2-image');
+                window.dataLayer.push({
+                    'data-ex-variant': 'experiment-hvt-visual-image',
+                    'data-ex-name': 'experiment-hvt-visual'
+                });
+            } else if (href.indexOf('v=animation') !== -1) {
+                utmSource = 'mozilla.org-firefox-welcome-8-animation';
+                document.body.classList.add('variation-3-animation');
+                window.dataLayer.push({
+                    'data-ex-variant': 'experiment-hvt-visual-animation',
+                    'data-ex-name': 'experiment-hvt-visual'
+                });
+            } else if (href.indexOf('v=header-text') !== -1) {
+                utmSource = 'mozilla.org-firefox-welcome-8-header-text';
+                var autopilotString = 'Your privacy is on <strong>autopilot</strong>';
+                if (navigator.language.startsWith('fr-')) {
```

Comment:
Rather than try to wrangle L10n in JS, it may be both easier and more performant to handle the variations in jinja / python on the server.

We have a pre-build helper just for experiments, that will pass things like `variation` into the view: https://bedrock.readthedocs.io/en/latest/coding.html#variation-views

Reply:
The reason this particular string is in JS is because I could not figure out how to set a string containing <strong> as the value for the title here:
```
{% call hero(
      title=ftl('welcome-page8-your-privacy-respected'),
      desc=ftl('welcome-page8-firefox-automatically-protects'),
      class='mzp-t-firefox mzp-t-dark variation-default',
      include_cta=True,
      heading_level=1
    ) %}
```

Reply:
Something like `title='Your privacy is on <strong>autopilot</strong>'|safe,` should work ok.

Reply:
is it possible to use a varibale in it's place, for translations? like: `title={{ autopilot }}|safe,` this gives a syntax error for me.

Reply:
Try:

```jinja
{% set autopilot ='Your privacy is on <strong>autopilot</strong>'|safe %}
```

and then:

```
title=autopilot
```

Reply:
excellent!

Reply:
We can also directly add translations in FTL files if we'd rather do that than hard code them in templates.

Reply:
> We can also directly add translations in FTL files if we'd rather do that than hard code them in templates.

Seeing as this is for an experiment, I think we'd probably want to hold off exposing new strings that might not make it past this PR.

Reply:
I guess it would unless we added an exception to the config. Makes sense.